code/game/objects/items/weapons/ranged/airgun.dm

```diff
@@ -0,0 +1,252 @@
+					playsound(src.loc, 'sound/foley/industrial/toggle.ogg', 100, FALSE)
+					loading_chamber = FALSE
+			else
+				to_chat(user, span_info("I begin to open the loading chamber..."))
+				if(do_after(user, use_time SECONDS, src))
+					to_chat(user, span_info("I open the loading chamber."))
+					playsound(src.loc, 'sound/foley/industrial/toggle.ogg', 100, FALSE)
+					loading_chamber = TRUE
+		if("Hand Crank")
+			if(cranked)
+				to_chat(user, span_info("I begin to turn the crank counter-clockwise..."))
+				if(do_after(user, use_time SECONDS, src))
+					to_chat(user, span_info("I turn the crank counter-clockwise, decompressing the spring."))
+					playsound(src.loc, 'sound/foley/winding.ogg', 100, FALSE)
+					cranked = FALSE
+			else
+				to_chat(user, span_info("I begin to turn the crank clockwise..."))
+				if(do_after(user, use_time SECONDS, src))
+					to_chat(user, span_info("I turn the crank clockwise, comporessing the spring."))
+					playsound(src.loc, 'sound/foley/winding.ogg', 100, FALSE)
+					cranked = TRUE
+		if("Steam Lever")
+			if(steam_lever)
+				to_chat(user, span_info("I begin to pull the steam lever down..."))
+				if(do_after(user, use_time SECONDS, src))
+					to_chat(user, span_info("I pull the steam lever down, disabling the flow of steam."))
+					playsound(src.loc, 'sound/foley/lock.ogg', 100, FALSE)
+					steam_lever = FALSE
+			else
+				to_chat(user, span_info("I begin to pull the steam lever up..."))
+				if(do_after(user, use_time SECONDS, src))
+					to_chat(user, span_info("I pull the steam lever up, enabling the flow of steam."))
+					playsound(src.loc, 'sound/foley/lock.ogg', 100, FALSE)
+					steam_lever = TRUE
+	update_appearance(UPDATE_ICON_STATE)
+	return SECONDARY_ATTACK_CANCEL_ATTACK_CHAIN
+
+/obj/item/gun/ballistic/revolver/grenadelauncher/airgun/process_fire(atom/target, mob/living/user, message = TRUE, params = null, zone_override = "", bonus_spread = 0)
+	if(user.client)
+		if(user.client.chargedprog >= 100)
+			spread = 0
+		else
+			spread = 150 - (150 * (user.client.chargedprog / 100))
+	else
+		spread = 0
+	if(!(cranked) || !(steam_lever) || (loading_chamber))
+		to_chat(user, span_warning("[src] refuses to fire!"))
+		playsound(src.loc, 'sound/foley/industrial/pneumatichiss.ogg', 100, FALSE)
+		return FALSE
+	for(var/obj/item/ammo_casing/CB in get_ammo_list(FALSE, TRUE))
+		var/obj/projectile/BB = CB.BB
+		if(user.client.chargedprog >= 100)
+			BB.accuracy += 15 //better accuracy for fully aiming
+		if(user.STAPER > 8)
+			BB.accuracy += (user.STAPER - 8) * 4 //each point of perception above 8 increases standard accuracy by 4.
+			BB.bonus_accuracy += (user.STAPER - 8) //Also, increases bonus accuracy by 1, which cannot fall off due to distance.
+		if(user.STAPER > 10)
+			BB.damage = BB.damage * (user.STAPER / 10) * (1 + pressure_to_use / 3) * pressure_to_use //2/3rds damage at pressure 1, normal at pressure 2, 4/3rds at pressure 3
```

Comment:
The damage is insane, Even with 11 PER i crit all the time and the dmg is just basically max all the time.

Pressure 3, 11 PER,

Look, 80 * 1/11, 1.1 , 88 * ( 3/3 + 1, = 2) so 176 * 3 = 528 like what.

This is the puffer formula:

BB.damage = BB.damage * (user.STAPER / 10) 

Air gun:

BB.damage = BB.damage * (user.STAPER / 10) * (1 + pressure_to_use / 3) * pressure_to_use

IT  is WAY stronger, besides being craftable.

Bruh..